src/crypto/elliptic_impl_priv.cpp

```diff
@@ -94,7 +94,7 @@ namespace fc { namespace ecc {
         do
         {
             FC_ASSERT( secp256k1_ecdsa_sign_compact( detail::_get_context(), (unsigned char*) digest.data(), (unsigned char*) result.begin() + 1, (unsigned char*) my->_key.data(), extended_nonce_function, &counter, &recid ));
-        } while( require_canonical && !public_key::is_canonical( result ) );
+        } while( !public_key::is_canonical( result, canon_type ) );
```

Comment:
@goldibex What about loop it self, is it okay to generate signature in the loop? There is a function secp256k1_ecdsa_signature_normalize which should do it.

you can find more here: https://github.com/bitcoin-core/secp256k1/blob/master/include/secp256k1.h#L461